virtool/db/mongo.py

```diff
@@ -74,3 +74,28 @@ async def get_mongo_version(db: AsyncIOMotorClient) -> str:
     :return: MongoDB server version in string format
     """
     return (await db.motor_client.client.server_info())["version"]
+
+
+async def init_check_db(db):
+
+    await db.analyses.create_index("sample.id")
+    await db.analyses.create_index([("created_at", -1)])
+    await db.caches.create_index([("key", 1), ("sample.id", 1)], unique=True)
+    await db.history.create_index("otu.id")
+    await db.history.create_index("index.id")
+    await db.history.create_index("created_at")
+    await db.history.create_index([("otu.name", 1)])
+    await db.history.create_index([("otu.version", -1)])
+    await db.indexes.create_index([("version", 1), ("reference.id", 1)], unique=True)
+    await db.keys.create_index("id", unique=True)
+    await db.keys.create_index("user.id")
+    await db.otus.create_index([
+        ("_id", pymongo.ASCENDING),
+        ("isolate.id", pymongo.ASCENDING)
+    ])
+    await db.otus.create_index("name")
+    await db.otus.create_index("nickname")
+    await db.otus.create_index("abbreviation")
+    await db.samples.create_index([("created_at", pymongo.DESCENDING)])
+    await db.sequences.create_index("otu_id")
+    await db.sequences.create_index("name")
```

Comment:
Where are those elite DRY skills?

You have the exact same lines in `virtool.startup.init_check_db`.

The `init_` functions are for initializing the application via `on_startup` and all take `app` as the only argument. I don't want to see that naming scheme used anywhere else.

* Extract logic to a new function `virtool.db.mongo.create_indexes(db)` and reuse in the testing fixture and in the real `init_check_db`.
* Document the new function. Unit testing it is complicated and not worth it.
* Remove blank line below function declaration